examples/nD_pyramid.py

```diff
@@ -0,0 +1,37 @@
+"""
+Displays an image pyramid
+"""
+
+from skimage import data
+from skimage.util import img_as_ubyte
+from skimage.color import rgb2gray
+from skimage.transform import pyramid_gaussian
+import napari
+from napari.util import app_context
+import numpy as np
+
+
+# create pyramid from astronaut image
+astronaut = rgb2gray(data.astronaut())
+base = np.tile(astronaut, (4, 4))
+base = np.array([base * (32 - i) / 32 for i in range(32)])
+print('base shape', base.shape)
+
+pyramid = list(pyramid_gaussian(base, downscale=2, max_layer=4,
+                                multichannel=False))
+pyramid = [img_as_ubyte(p) for p in pyramid]
+print('pyramid level shapes: ', [p.shape for p in pyramid])
+
+with app_context():
+    # create the viewer
+    viewer = napari.Viewer()
+
+    # add image pyramid
+    viewer.add_pyramid(pyramid, clim_range=[0, 255])
+
+    # Set the view box of the camera to include the whole base image of the
+    # pyramid with a little padding. The view box is a 4-tuple of the x, y
+    # corner position followed by width and height
+    base_shape = pyramid[0].shape[1:]
+    viewer.camera.rect = (-0.1 * base_shape[1], -0.1 * base_shape[0],
+                          1.2 * base_shape[1], 1.2 * base_shape[0])
```

Comment:
Continuing the [discussion](https://github.com/napari/napari/pull/295/files#r289509749) from the original pyramid PR:

> Actually I'm wary of doing this automatically as there could be times when you add a layer and you don't want the camera to change

I think this is a general problem for napari that we have somehow avoided thus-far. In the case of adding a 2D image, you probably don't want the camera to change *if you've already interacted with the camera*, or at least *if there is already a previous layer on the canvas*. I think *in general*, `add_layer` should set the camera to fully contain the first layer it gets.

That's my proposal anyway. Either way, I think the code above has to go — it's clunky to begin with, but the fact that the coordinates are in xy order makes it even worse! I don't think this is something we want in our examples!

... Although there is a middle road: add a function `viewer.camera.set_view_corners`, which uses data coordinates.

Reply:
I can fix this, I will add a `camera_viewbox` property on the base layer and viewer and use events to propogate any changes from it up to the canvas. The `Pyramid` layer will then be able to set the camera when it is added (the viewer could then do this too if preferred)

Reply:
actually, that wouldn't have been so good as the property would have gotten out of sync pretty quickly. instead i've adapted the `reset_view` event to now include the shape of the `viewbox` that will make everything look good. the unpleasant lines in all the examples have been removed